Ft roles register drivers operators

Byadiso opened the pull request:
#### What does this PR do?
Register Drivers and Operators, permissions, and roles.

####  Description of Task to be completed?
Created Signup API.
Created API to get all users.
Created API for getting a single user by ID.
Created API for deleting a sing user.
Created API for Updating a single user.
Created API for Updating roles for operator, driver, and administrator.
Implemented password Autogeneration.
wip

####  How should this be manually tested?
step 1: clone git repo if not yet done.
step 2: cd Avengers-Phantom-BE.
step 3: npm i or npm install to install dependencies.
step 4: npm run migrate to migrate database.
step 5: npm run dev to start the server.

#### APIS

api/v1/users/signup.
api/v1/users all users.
api/v1/users/:uuid update user details.
api/v1/users/:uuid update role.
api/v1/users/:uuid delete.
api/v1/users/:uuid get a use.

Commit:
ft(roles register and permissions): implementing permissions features

Updated routes and controllers from users files
removed duplicate in app.js file

Added CI and initial  test

Create phantom.yml

added first  unit test

Added CI and initial test

adding-sinon-setup

Added CI and initial test

rebase

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip
signin to phantom

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

updated

ft(registration):register drivers and operators

created signup API
created user route
created get all users API
created get user API
created update user API
created delete user API
wip

ft(signin to phantom):sign user to phantom

created user signin functionality APIs
wip

ft(signin):signin to phantom

hashed password
wip

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip
signin to phantom

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

updated

ft(registration):register drivers and operators

created signup API
created user route
created get all users API
created get user API
created update user API
created delete user API
wip

ft(signin to phantom):sign user to phantom

created user signin functionality APIs
wip

ft(signin):signin to phantom

hashed password
wip

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

updated

ft(registration):register drivers and operators

created signup API
created user route
created get all users API
created get user API
created update user API
created delete user API
wip

ft(signin to phantom):sign user to phantom

created user signin functionality APIs
wip

ft(signin):signin to phantom

hashed password
wip

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(Create roles and permissions)
Created roles model
wip

ft(adding validation on empty and duplicate roles)

ft(roles,permission):roles and permissions

created roles models
created roles controllers
created  routes route
associated permissions to roles
created middlewares to utilize our predefined permissions
wip

ch(Update the way importing modules for test)
remove Es6 for to allow test passing
wip

ft(roles an permissions):updates
updated
wip

ft(adding-validation-uniqueness)

ft(adding update roles middeware)

ft(roles-permission): Setting up roles and permissions

ft(adding update roles middeware)

ft(roles-permission): Setting up roles and permissions

rebase

ft(roles, permission, register driver and operator)
register driver
register operator
roles for admin
roles for operator
roles for driver
their permissions

Commit:
ft(roles register and permissions): implementing permissions features

Updated routes and controllers from users files
removed duplicate in app.js file

Added CI and initial  test

Create phantom.yml

added first  unit test

Added CI and initial test

adding-sinon-setup

Added CI and initial test

rebase

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip
signin to phantom

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

updated

ft(registration):register drivers and operators

created signup API
created user route
created get all users API
created get user API
created update user API
created delete user API
wip

ft(signin to phantom):sign user to phantom

created user signin functionality APIs
wip

ft(signin):signin to phantom

hashed password
wip

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip
signin to phantom

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

updated

ft(registration):register drivers and operators

created signup API
created user route
created get all users API
created get user API
created update user API
created delete user API
wip

ft(signin to phantom):sign user to phantom

created user signin functionality APIs
wip

ft(signin):signin to phantom

hashed password
wip

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(seeders):Generated Dummy user in seeders

updated

ft(registration):register drivers and operators

created signup API
created user route
created get all users API
created get user API
created update user API
created delete user API
wip

ft(signin to phantom):sign user to phantom

created user signin functionality APIs
wip

ft(signin):signin to phantom

hashed password
wip

ch(swagger set up): Setting up swagger

Updated swagger.js file
Added configurations inside swagger.js file
Connected swagger documentation to our express app
wip

ft(postgres DB): Added Dummy user Table

created user model
created post request
created get request for all users
created get request for a single user with user id
created delete user request
Done

ft(Create roles and permissions)
Created roles model
wip

ft(adding validation on empty and duplicate roles)

ft(roles,permission):roles and permissions

created roles models
created roles controllers
created  routes route
associated permissions to roles
created middlewares to utilize our predefined permissions
wip

ch(Update the way importing modules for test)
remove Es6 for to allow test passing
wip

ft(roles an permissions):updates
updated
wip

ft(adding-validation-uniqueness)

ft(adding update roles middeware)

ft(roles-permission): Setting up roles and permissions

ft(adding update roles middeware)

ft(roles-permission): Setting up roles and permissions

rebase

ft(roles, permission, register driver and operator)
register driver
register operator
roles for admin
roles for operator
roles for driver
their permissions

Byadiso committed:
ft(roles, permission, register driver and operator)
register driver
register operator
roles for admin
roles for operator
roles for driver
their permissions

Jeanndo reviewed (changes requested):
maybe we can find a way of putting config.js variables in environment variables

Jeanndo reviewed (commented):
maybe we can put these variables in a dotenv file

abraham-kwizera reviewed (commented):
all test are not passing due to missing package-lock.json fille, 
you may add it too.

Byadiso committed:
ft(roles, permission, register driver and operator)
register driver
register operator
roles for admin
roles for operator
roles for driver
their permissions

Byadiso committed:
ft(roles-drivers-operators)
